src/sign_tx.c

```diff
@@ -0,0 +1,449 @@
+        // button pressed after it's been already confirmed,
+        // while processing signatures. Just ignore it.
+        return 0;
+    }
+
+    //TODO maybe we want the user to hold either button instead of a simple click
+    switch (button_mask) {
+        case BUTTON_LEFT:
+        case BUTTON_EVT_FAST | BUTTON_LEFT: // cancel
+            io_exchange_with_code(SW_USER_REJECTED, 0);
+            // Return to the main screen.
+            ui_idle();
+            break;
+
+        case BUTTON_RIGHT:
+        case BUTTON_EVT_FAST | BUTTON_RIGHT: // confirm
+            ctx->state = USER_APPROVED;
+            io_exchange_with_code(SW_OK, 0);
+            os_memmove(ctx->line1, "Processing\0", 11);
+            os_memmove(ctx->line2, "...\0", 4);
+            UX_REDISPLAY();
+            break;
+    }
+    return 0;
+}
+
+// Define the sign tx screen. User will be able to scroll through an output.
+// (address + value) with left/right buttons. When he's done, he will click both
+// buttons and see next output. A final confirmation screen appears before 
+// sending tokens.
+static const bagl_element_t ui_sign_tx_compare[] = {
+    UI_BACKGROUND(),
+
+    // Left and right buttons for scrolling the text.
+    UI_ICON_LEFT(0x01, BAGL_GLYPH_ICON_LEFT),
+    UI_ICON_RIGHT(0x02, BAGL_GLYPH_ICON_RIGHT),
+
+    UI_TEXT(0x00, 0, 12, 128, global.sign_tx_context.line1),
+    UI_TEXT(0x00, 0, 26, 128, global.sign_tx_context.line2),
+};
+
+// Preprocessor for this screen. Hides left or right arrows depending on the
+// scrolling position.
+static const bagl_element_t* ui_prepro_sign_tx_compare(const bagl_element_t *element) {
+    switch (element->component.userid) {
+    case 1:
+        // 0x01 is the left icon so return NULL if we're displaying the beginning of the text.
+        return (ctx->display_index == 0) ? NULL : element;
+    case 2:
+        // 0x02 is the right, so return NULL if we're displaying the end of the text.
+        return ctx->display_index == (strlen((const char*)ctx->info) - 12) ? NULL : element;
+    default:
+        // Always display all other elements.
+        return element;
+    }
+}
+
+// This is the button handler for the outputs screen.
+static unsigned int ui_sign_tx_compare_button(unsigned int button_mask, unsigned int button_mask_counter) {
+    switch (button_mask) {
+        // scroll left by either clicking or pressing the left button
+        case BUTTON_LEFT:
+        case BUTTON_EVT_FAST | BUTTON_LEFT: // SEEK LEFT
+            if (ctx->display_index != 0) {
+                ctx->display_index--;
+                os_memmove(ctx->line2, ctx->info + ctx->display_index, 12);
+                UX_REDISPLAY();
+            }
+
+            break;
+
+        // scroll right by either clicking or pressing the right button
+        case BUTTON_RIGHT:
+        case BUTTON_EVT_FAST | BUTTON_RIGHT: // SEEK RIGHT
+            if (ctx->display_index != strlen((const char*)ctx->info) - 12) {
```

Comment:
Replace 12 by the constant.

Reply:
There's a comment already to replace it after the other PR is merged.